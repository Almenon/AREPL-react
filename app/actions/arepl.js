// @flow
import type { counterStateType } from '../reducers/counter';
import {PythonEvaluator} from 'arepl-backend'

let myPython = new PythonEvaluator();
// initially pythonEvalFolderPath is assigned to __dirname before constructor is even called
// in which case it is just / which mucks things up
myPython.pythonEvalFolderPath = process.cwd() + "\\node_modules\\arepl-backend\\python"
myPython.startPython();

type actionType = {
  +type: string
};

export const PRINT_RESULT = 'PRINT_RESULT';
export const EXECUTION_COMPLETE = 'EXECUTION_COMPLETE';

export function printResult(printLine:string) {
  return {
    type: PRINT_RESULT,
    printLine:printLine
  };
}

// flow SHOULD have picked up on result type from arepl but is acting wierd
export function executionComplete(result: any) {
  result['type'] = EXECUTION_COMPLETE;
  return result;
}

export function executeCode(code: {savedCode:string,evalCode:string}) {
    return (dispatch: (action: actionType) => void) => {
        
      myPython.onPrint = (printLine)=>{
        dispatch(printResult(printLine));
      }
      myPython.onResult = (result)=>{
          dispatch(executionComplete(result));
      }

      myPython.execCode(code);
      
    };
}
