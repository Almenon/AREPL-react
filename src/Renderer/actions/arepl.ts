import {PythonEvaluator} from 'arepl-backend'

const myPython = new PythonEvaluator();
// initially pythonEvalFolderPath is assigned to __dirname before constructor is even called
// in which case it is just / which mucks things up
// @ts-ignore
myPython.pythonEvalFolderPath = process.cwd() + "\\node_modules\\arepl-backend\\python"
myPython.startPython();

interface actionType{
  type: string
};

export const PRINT_RESULT = 'PRINT_RESULT';
export const EXECUTION_COMPLETE = 'EXECUTION_COMPLETE';

export function printResult(printLine:string) {
  return {
    "type": PRINT_RESULT,
    "printLine":printLine
  };
}

export function executionComplete(result: any) {
  result['type'] = EXECUTION_COMPLETE;
  return result;
}

export function executeCode(code: {savedCode:string,evalCode:string}) {
    return (dispatch: (action: actionType) => void) => {
      myPython.debounce(()=>{
        myPython.onPrint = (printLine)=>{
          dispatch(printResult(printLine));
        }
        myPython.onResult = (result)=>{
            dispatch(executionComplete(result));
        }
  
        myPython.execCode(code);
      }, 400)
    };
}
