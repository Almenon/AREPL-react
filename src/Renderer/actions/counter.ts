import { RootState } from "@red/index";

export type counterActions = increment | decrement;

interface increment {
  type: string;
}

interface decrement {
  type: string;
}

const increment = () => ({
  type: "INCREMENT_COUNTER"
});

const decrement = () => ({
  type: "DECREMENT_COUNTER"
});

const incrementIfOdd = () => {
  return (
    dispatch: (action: counterActions) => void,
    getState: () => RootState
  ) => {
    const { counter } = getState().counter;

    if (counter % 2 === 0) {
      return;
    }

    dispatch(increment());
  };
}

const incrementAsync = (delay: number = 1000) => {
  return (dispatch: (action: counterActions) => void) => {
    setTimeout(() => {
      dispatch(increment());
    }, delay);
  };
};

export const actionCreators = {
  increment,
  decrement,
  incrementIfOdd,
  incrementAsync
};