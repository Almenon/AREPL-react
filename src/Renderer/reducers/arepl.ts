// @flow
import { PRINT_RESULT, EXECUTION_COMPLETE } from '../actions/arepl';

interface actionType {
  type: string
};

export default function arepl(state: any, action: actionType) {
  switch (action.type) {
    case PRINT_RESULT:
      return state;
    case EXECUTION_COMPLETE:
      return state;
    default:
      return state;
  }
}
