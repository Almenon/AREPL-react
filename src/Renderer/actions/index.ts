import { counterActions } from "@act/counter";

export type RootActions = counterActions;