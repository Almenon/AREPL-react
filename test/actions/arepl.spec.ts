import { executeCode } from '../../src/Renderer/actions/arepl'

describe('arepl actions', () => {
    it('should evaluate simple python code', (done)=>{

        const codeExecutor = executeCode({savedCode:"",evalCode:"x=1+1"})

        const dispatch = (result) => {

            // clear these because they may vary randomly
            result.execTime = 0
            result.totalTime = 0
            result.totalPyTime = 0

            expect(result).toMatchSnapshot()
            done()
        }

        codeExecutor(dispatch)
    })
})