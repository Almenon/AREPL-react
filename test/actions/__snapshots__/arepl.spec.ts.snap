// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arepl actions should evaluate simple python code 1`] = `
Object {
  "ERROR": "",
  "execTime": 0,
  "totalPyTime": 0,
  "totalTime": 0,
  "type": "EXECUTION_COMPLETE",
  "userVariables": Object {
    "x": 2,
  },
}
`;
